src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2776,9 +2777,10 @@ private BoundExpression BindIsOperator(BinaryExpressionSyntax node, DiagnosticBa
                 return new BoundIsOperator(node, operand, typeExpression, Conversion.NoConversion, resultType, hasErrors: true);
             }
 
-            if (node.Right is IdentifierNameSyntax name && name.Identifier.Text == "_")
+            bool recursivePatternsEnabled = ((CSharpParseOptions)node.SyntaxTree.Options).IsFeatureEnabled(MessageID.IDS_FeatureRecursivePatterns);
+            if (recursivePatternsEnabled && wasUnderscore)
```

Comment:
>recursivePatternsEnabled && wasUnderscore [](start = 16, length = 41)

Consider writing as `wasUnderscore && IsFeatureEnabled(...)` to avoid calling `IsFeatureEnabled` unnecessarily.